test/unit/serverless/aws-lambda.test.js

```diff
@@ -337,26 +322,26 @@ describe('AwsLambda.patchLambdaHandler', () => {
       function confirmAgentAttribute(transaction) {
         const agentAttributes = transaction.trace.attributes.get(ATTR_DEST.TRANS_EVENT)
 
-        expect(agentAttributes).to.not.have.property('request.headers.X-Amz-Cf-Id')
-        expect(agentAttributes).to.not.have.property('request.headers.X-Forwarded-For')
-        expect(agentAttributes).to.not.have.property('request.headers.X-Forwarded-Port')
-        expect(agentAttributes).to.not.have.property('request.headers.X-Forwarded-Proto')
+        t.not('request.headers.X-Amz-Cf-Id' in agentAttributes)
+        t.not('request.headers.X-Forwarded-For' in agentAttributes)
+        t.not('request.headers.X-Forwarded-Port' in agentAttributes)
+        t.not('request.headers.X-Forwarded-Proto' in agentAttributes)
 
-        expect(agentAttributes).to.not.have.property('request.headers.xAmzCfId')
-        expect(agentAttributes).to.not.have.property('request.headers.xForwardedFor')
-        expect(agentAttributes).to.not.have.property('request.headers.xForwardedPort')
-        expect(agentAttributes).to.not.have.property('request.headers.xForwardedProto')
+        t.not('request.headers.xAmzCfId' in agentAttributes)
+        t.not('request.headers.xForwardedFor' in agentAttributes)
+        t.not('request.headers.xForwardedPort' in agentAttributes)
+        t.not('request.headers.xForwardedProto' in agentAttributes)
 
-        expect(agentAttributes).to.not.have.property('request.headers.XAmzCfId')
-        expect(agentAttributes).to.not.have.property('request.headers.XForwardedFor')
-        expect(agentAttributes).to.not.have.property('request.headers.XForwardedPort')
-        expect(agentAttributes).to.not.have.property('request.headers.XForwardedProto')
+        t.not('request.headers.XAmzCfId' in agentAttributes)
```

Comment:
I think these `t.not` assertions are giving a false positive passing test. `t.notOk('agentAttributes.request.headers.XamzCfId')` should work.